### DISCUSSION TITLE
nvim gets stuck on the specific completion option in popup

### DISCUSSION THREAD

**Discussion by justjew** [LazyVim/LazyVim#43] [Q&A] [answered]
When trying to type a keyword `const` in a `.dart` file, as soon as cmp suggests the `consolidateHttpClientResponseBytes(...)` option in popup, nvim gets stuck forever. Only terminal closing can help.

Tried to clean cache, state etc, not helped.
Logs are clear, no errors shown.
The only additional package I've installed is flutter-tools.nvim

**Comment by folke** [comment, marked as answer]
That's odd. Can you update LazyVim and double check the problem still occurs?

**Reply by justjew** [reply]
All up to date

**Reply by folke** [reply]
Can you add `{"folke/noice.nvim", enabled = false}` to one of your specs to see if that fixes it?

I know there's a endless loop somwehere in the Neovim treesitter code. It might trigger somehow for dart.

**Reply by justjew** [reply]
Yes, it did the trick

**Reply by folke** [reply]
damn it. ok. it's a bug I was planning to fix upstream in Neovim. Will do so soon :)

**Reply by justjew** [reply]
Thanx, friend! I'll wait for it

**Reply by justjew** [reply]
> damn it. ok. it's a bug I was planning to fix upstream in Neovim. Will do so soon :)

not fixed yet? 
I just checked, it still crashes if `noice.nvim` is enabled

**Reply by folke** [reply]
nope, not fixed yet. It's a bug in the Neovim treesitter code. Basically what happens is that some languages define embedded languages recursively. I don't have this issue myself with what I use, so this is low on my priority list to fix it in Neovim.

**Reply by folke** [reply]
So I looked into this last night and this morning and unfortunately, the issue seems to be in [treesitter](https://github.com/tree-sitter/tree-sitter) itself. Sometimes the parser just never returns and keeps consuming all memory.

Not something I can fix. There's a ton of related issues in the treesitter repo.

**Reply by folke** [reply]
Good news! I found that the latest treesitter version already fixes this.
Once https://github.com/neovim/neovim/pull/21858 gets merged, Neovim won't crash anymore.

**Reply by justjew** [reply]
Awesome news! Gonna check it. Thanks

**Reply by justjew** [reply]
Oh, bad news.
I updated `nvim` to `v0.9.0-dev-758+gd8d39344e-dirty`.
It still crashes, but in a bit different way.

While I am typing `const` keyword, the popup appears, but the crashing option `consolidateHttpClientResponseBytes(...)` is now at second position. If I press `C-n` to move to it, `nvim` crashes

**Reply by justjew** [reply]
`consolidateHttpClientResponseBytes` is the only option which make nvim crash

**Reply by folke** [reply]
Too bad, probably another Treesitter bug that is still in there...

I do note you have a `dirty` build. Maybe try doing a clean build of master just to be sure?

**Reply by justjew** [reply]
The same on `v0.9.0-dev-757+g51d082b46`
